### PR TITLE
Option for Custom storage

### PR DESCRIPTION
**Pull request by pavankumarkatakam**
Other than file, PDO and redis, This PR add the functionality to developers to have their own Storage interface.

Working:
in config 

     'storage' => array(
        'enabled' => true,
        'driver' => 'custom', // redis, file, pdo, custom
        'path' => storage_path() . '/debugbar', // For file driver
        'connection' => null,   // Leave null for default connection (Redis/PDO)
        'provider'=>"App/Framework/ElasticStorage" // Instance of StorageInterface for custom driver
    ),

    <?php

  class ElasticStorage implements StorageInterface{
 
  }

### CODE REVIEW

**Comment by pavankumarkatakam** [comment]
Here the Storage implemented class can inject services like router, storage .... etc.

File: config/debugbar.php [hunk 1]
```diff
@@ -28,9 +28,10 @@
      */
     'storage' => array(
         'enabled' => true,
-        'driver' => 'file', // redis, file, pdo
+        'driver' => 'file', // redis, file, pdo, custom
         'path' => storage_path() . '/debugbar', // For file driver
         'connection' => null,   // Leave null for default connection (Redis/PDO)
+        'provider'=>"" // Instance of StorageInterface for custom driver
```
Comment (barryvdh) [inline config/debugbar.php]: Can you follow CS? Spacing and quotes, like the rest of the options?

File: src/LaravelDebugbar.php [hunk 2]
```diff
@@ -785,6 +785,10 @@ protected function selectStorage(DebugBar $debugbar)
                     $connection = $config->get('debugbar.storage.connection');
                     $storage = new RedisStorage($this->app['redis']->connection($connection));
                     break;
+                case 'custom':
+                    $class = $config->get('debugbar.storage.provider');
+                    $storage = app($class);
```
Comment (barryvdh) [inline src/LaravelDebugbar.php]: I prefer `$this->app->make($provider)`

**Comment by barryvdh** [comment]
Looks interesting, but why do you need extra storage?

**Comment by barryvdh** [comment]
I'm guessing for #438 then?

**Comment by barryvdh** [comment]
If you have good storage created, please consider adding it via a PR to https://github.com/maximebf/php-debugbar/tree/master/src/DebugBar/Storage

**Comment by pavankumarkatakam** [comment]
Yes @barryvdh that issue created by me only, Actually I need custom storage to get the control over the  data that is generated while storing so that i can achieve my requirement. I didn't say that elasticsearch is the best but for some reasons i want to use the ES. And may be some other developer want to have some storage rather than creating all the storage implementations if there is option for developers to create own storage then every developer can have their own way of storing the data and of course the file is default. So rather than integrating the ElasticStorage to package give the option for developers to develop their own.

**Comment by pavankumarkatakam** [comment]
And for your information , I'm using the data generated by this package for custom analytics. Like 
1) How fast the app  is running?
2)  How many users are visiting my site?
3) What is the error rate and response code?

**Comment by barryvdh** [comment]
Thanks!

**Comment by pavankumarkatakam** [comment]
I think we want to mention in readme.

**Comment by pavankumarkatakam** [comment]
Hi @barryvdh , may i know when you are going to release the master branch.

**Pull request merged**